build/templates/library_singleton.py.mako

```diff
@@ -35,10 +38,19 @@ def get():
     Returns the library.Library singleton for ${config['module_name']}.
     '''
     global _instance
-    if _instance is None:
-        try:
-            _instance = library.Library(_get_library_name(), _get_library_type())
-        except OSError:  # pragma: no cover
-            raise errors.DriverNotInstalledError()
-    return _instance
+    global _instance_lock
+
+    with _instance_lock:
+        if _instance is None:
+            try:
+                library_type = _get_library_type()
+                if library_type == 'windll':
+                    ctypes_library = ctypes.WinDLL(_get_library_name())
+                else:  # pragma: no cover
+                    assert library_type == 'cdll'
+                    ctypes_library = ctypes.CDLL(_get_library_name())
+                    _instance = library.Library(ctypes_library)
```

Comment:
I think this needs to be outdented by 1 level.

Reply:
Also, remove TODO from PR description.

Reply:
fixed